### PR TITLE
Support new mate-menus API

### PR DESCRIPTION
**Pull request by yetist**
Before that, I submitted a [PR](https://github.com/mate-desktop/mate-menus/pull/59) for mate-menus, if that PR merged, it will break mate-screensaver.

So now I've changed the mate-screensaver code to use the new api of mate-menus 1.21.0.

### CODE REVIEW

**Comment by raveit65** [comment]
Which part of gui needs to tested?
Or where can i see changes?
Or does everything works in the background?

**Comment by monsta** [comment]
Do we accept version bumps here, or do we add these ourselves after merging?

File: src/gs-theme-manager.c [hunk 1]
```diff
@@ -397,12 +378,19 @@ static MateMenuTree *
...
 {
 	MateMenuTree *themes_tree = NULL;
+	GError *error = NULL;
 
 	/* we only need to add the locations to the path once
 	   and since this is only run once we'll do it here */
 	add_known_engine_locations_to_path ();
 
-	themes_tree = matemenu_tree_lookup ("mate-screensavers.menu", MATEMENU_TREE_FLAGS_NONE);
+	themes_tree = matemenu_tree_new ("mate-screensavers.menu", MATEMENU_TREE_FLAGS_SORT_DISPLAY_NAME);
```
Comment (monsta) [inline src/gs-theme-manager.c]: Seems like this sorting isn't used anywhere in the code, so probably there's no need to add it.

**Comment by raveit65** [comment]
Version bumping or editing NEWS files should be done from release team in general, as it depends on other things. And i see no package which needs to be rebuild because mate-screensaver.
@yetist 
Can you please revert this?

**Comment by raveit65** [comment]
Same in other PRs, we all know that we have to bump build requires to match new mate-menus.

File: src/gs-theme-manager.c (hunk 1 above)
Comment (yetist) [reply src/gs-theme-manager.c]: The screensaver was not sorted before, and it's ok now. Don't you like it?

Comment (monsta) [reply src/gs-theme-manager.c]: Maybe I missed the sorting - I didn't notice any differences. Where should I look?

**Comment by monsta** [comment]
Ok, version bump is left in mate-menus PR. If it's removed, then the build requirement bump here has to go as well.

**Comment by monsta** [comment]
Looks ok, I'm just a bit confused abort sorting (see my comments above).

**Comment by yetist** [comment]
Sorted screensaver themes here:

**Comment by monsta** [comment]
@raveit65: the last commit reverts the menu sorting on loading, which doesn't seem to affect anything (the screensaver list is sorted somewhere else apparently).

**Comment by raveit65** [comment]
Squashed all in one commit and merged.
https://github.com/mate-desktop/mate-screensaver/commit/c3807e72fc761778100f500c0966fe491aec7fd2
Thanks for your work.

**Comment by raveit65** [comment]
mate-screensaver-1.21 is released but not tagged.
Tarball is at pre-release folder for testing.

**Pull request closed without merging**